python/src/vmaf/core/noref_feature_extractor.py

```diff
@@ -431,3 +431,67 @@ def mscn_extract_niqe(cls, img, patch_size, mode):
             assert False
 
         return list_features
+
+
+class SiTiNorefFeatureExtractor(NorefExecutorMixin, FeatureExtractor):
```

Comment:
maybe we can add a disclaimer that this implementation makes a number of certain assumptions (e.g. the 75 percentile) and that other SI-TI implementations are available too?

Reply:
Not sure. For now I think we are OK and we should validate it against BT.500-14 Annex 6 at some point of time.

Reply:
Sorry for dropping in on this thread, but are there any test vectors for SI/TI at all? Annex 6 of BT.500 only specifies the equations.

I think your implementation is functionally equivalent to what I did [here](https://github.com/slhck/siti), so I guess it's fine. There's also an [OpenCV implementation](https://github.com/Telecommunication-Telemedia-Assessment/SITI/).

Reply:
@slhck as you said, I don't have any test vectors. Are you aware of any? At least I plan to run through the equations to make sure the math matches.

Reply:
No, I am not aware of any. Maybe we can cross-check with Margaret at NTIA if she has something from earlier VQEG FR tests. If all our three implementations matched, that would also be good. (I *did* cross-check the OpenCV code with my Python code though.)